### PR TITLE
feature/PATRO23-340 - add hangfire background job

### PR DESCRIPTION
**Pull request by DawidBski**
https://tracker.intive.com/jira/browse/PATRO23-340

### CODE REVIEW

File: src/api/app/Filters/DashboardAuthorizationFilter.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+			var token = hand.ReadJwtToken(accessToken);
+			string realmAccessValue = token.Claims.First(c => c.Type == "realm_access").Value;
+			if (!string.IsNullOrEmpty(this.role) && !realmAccessValue.Contains(this.role))
+			{
+				return false;
+			}
+		}
+		catch (Exception e)
+		{
+			Console.WriteLine(e);
```
Comment (mateuszgiedzinski) [inline src/api/app/Filters/DashboardAuthorizationFilter.cs:66]: use logger instead.

File: src/api/app/Configuration/HangfireConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	/// Configures Hangfire Dashboard and adds it to the IApplicationBuilder.
+	/// </summary>
+	/// <param name="app">The IApplicationBuilder to configure Hangfire Dashboard on.</param>
+	public static void UseHangfireService(this IApplicationBuilder app)
+	{
+		app.UseHangfireDashboard("/hangfire", new DashboardOptions
+		{
+			DashboardTitle = "Patronage Hangfire Dashboard",
+			IsReadOnlyFunc = (Func<DashboardContext, bool>)(_ => true),
+			Authorization = new[] { new DashboardAuthorizationFilter() },
```
Comment (mateuszgiedzinski) [inline src/api/app/Configuration/HangfireConfiguration.cs:38]: inject dashboard authorization filter

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

**Review by Kepeczsz — APPROVED** [review]

**Review by KlaudiaGajocha — APPROVED** [review]

File: src/api/app/appsettings.Development.json [hunk 3]
```diff
@@ -2,12 +2,14 @@
...
-			"Microsoft.AspNetCore": "Warning"
+			"Microsoft.AspNetCore": "Warning",
+			"Hangfire": "Information"
 		}
 	},
 	"ConnectionStrings": {
 		"AppDb": "Server=db; Database=IntivePatronage-2023-DB; User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true",
-		"DockerDb": "Server=localhost,1433; Database=IntivePatronage-2023-DB; User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true"
+		"DockerDb": "Server=localhost,1433; Database=IntivePatronage-2023-DB; User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true",
+		"HangfireConnection": "Server=db;Database=Hangfire;User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true; Integrated Security=false"
```
Comment (mateuszgiedzinski) [inline src/api/app/appsettings.Development.json:12]: why not localhost,1433?

File: src/modules/example/api/Controllers/ExampleController.cs [hunk 4]
```diff
@@ -55,6 +58,9 @@ public async Task<IActionResult> GetExamples([FromQuery] GetExamples request)
 		if (validationResult.IsValid)
 		{
 			var pagedList = await this.queryBus.Query<GetExamples, PagedList<ExampleInfo>>(request);
+
+			BackgroundJob.Enqueue(() => Console.WriteLine("Sent From The Background Task"));
```
Comment (mateuszgiedzinski) [inline src/modules/example/api/Controllers/ExampleController.cs:62]: I suggest not to use Console directly.

File: src/api/app/Configuration/HangfireConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+			IsReadOnlyFunc = (Func<DashboardContext, bool>)(_ => true),
+			Authorization = new[] { dashboardAuthorizationFilter },
+		});
+
+		RecurringJob.AddOrUpdate(
+			"SuperImportantJobId",
+			() => Console.WriteLine("Hello did you forget about me??"),
+			Cron.Hourly);
+
+		BackgroundJob.Schedule(() => Console.WriteLine("Schedule Job"), TimeSpan.FromDays(1));
```
Comment (mateuszgiedzinski) [inline src/api/app/Configuration/HangfireConfiguration.cs:47]: I suggest not use Console directly.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: src/api/app/appsettings.Development.json (hunk 3 above)
Comment (DawidBski) [reply src/api/app/appsettings.Development.json:12]: Because I Created Database in DockerDb. I tested it and it only working that way.

**Review by mateuszmuszynski — APPROVED** [review]

**Review by mateuszgiedzinski — APPROVED** [review]

**Pull request merged**